### PR TITLE
Fetch favorites in favorites screen

### PR DESCRIPTION
**Pull request by AbdulrahmanAli19**

### CODE REVIEW

**Comment by AbdulrahmanAli19** [comment]
is done with its owne logic.
NOTE ##You should reinstall the app because there's a new variable has been added to FavoriteProduct.kt entity

File: app/src/main/java/com/iti/android/team1/ebuy/model/datasource/repository/Repository.kt [hunk 1]
```diff
@@ -93,10 +92,13 @@ class Repository(
         localSource.removeAllFavoriteProducts()
     }
 
-    override suspend fun addProductToFavorite(product: Product): DatabaseResponse<Long> {
+    override suspend fun addProductToFavorite(
+        product: Product,
+        noOfItems: Int,
```
Comment (MohabElziny) [inline app/src/main/java/com/iti/android/team1/ebuy/model/datasource/repository/Repository.kt:97]: set default value in the repository for noOfItems

File: app/src/main/java/com/iti/android/team1/ebuy/ui/savedItems/view/SavedItemsFragment.kt [hunk 2]
```diff
@@ -1,27 +1,131 @@
...
     }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        viewModel.getFavoritesList()
+        fetchFavoritesData()
+    }
+
+    private fun fetchFavoritesData() {
+        lifecycleScope.launch {
```
Comment (MohabElziny) [inline app/src/main/java/com/iti/android/team1/ebuy/ui/savedItems/view/SavedItemsFragment.kt:56]: use launchWhenStarted to be aware of lifecycle while using stateflow

File: app/src/main/java/com/iti/android/team1/ebuy/ui/savedItems/view/SavedItemsFragment.kt [hunk 3]
```diff
@@ -1,27 +1,131 @@
...
+
+    private var onIncreaseClick: (FavoriteProduct, Int) -> Unit = { favoriteProduct, position ->
+        viewModel.updateFavoriteProduct(favoriteProduct)
+        fetchUpdateState(true, favoriteProduct, position)
+
+    }
+
+    private var onDecreaseClick: (FavoriteProduct, Int) -> Unit = { favoriteProduct, position ->
+        viewModel.updateFavoriteProduct(favoriteProduct)
+        fetchUpdateState(false, favoriteProduct, position)
```
Comment (MohabElziny) [inline app/src/main/java/com/iti/android/team1/ebuy/ui/savedItems/view/SavedItemsFragment.kt:104]: Is it right to lunch the coroutine every time I press the button?

File: app/src/main/java/com/iti/android/team1/ebuy/ui/savedItems/view/SavedItemsFragment.kt [hunk 4]
```diff
@@ -1,27 +1,131 @@
...
+    }
+
+    private fun fetchUpdateState(
+        isIncrease: Boolean,
+        favoriteProduct: FavoriteProduct,
+        position: Int,
+    ) {
+        lifecycleScope.launch {
```
Comment (MohabElziny) [inline app/src/main/java/com/iti/android/team1/ebuy/ui/savedItems/view/SavedItemsFragment.kt:112]: use launchWhenStarted to be aware of lifecycle while using stateflow

File: app/src/main/java/com/iti/android/team1/ebuy/model/datasource/repository/Repository.kt (hunk 1 above)
Comment (AbdulrahmanAli19) [reply app/src/main/java/com/iti/android/team1/ebuy/model/datasource/repository/Repository.kt:97]: An overriding function is not allowed to specify default values for its parameters, which means it is not possible to give a default value in the repository

**Review by MohabElziny — DISMISSED** [review]

**Review by AhmedAbdallah125 — APPROVED** [review]

File: app/src/main/java/com/iti/android/team1/ebuy/ui/savedItems/adapter/SavedRecyclerAdapter.kt [hunk 5]
```diff
@@ -1,27 +1,73 @@
...
+            binding.parent.setOnClickListener {
+                onItemClick(favorites[bindingAdapterPosition].productID)
+            }
+
+            binding.likeBtn.setOnLikeListener(object : OnLikeListener {
+                override fun liked(likeButton: LikeButton?) {}
 
-    override fun onBindViewHolder(holder: SavedItemsViewHolder, position: Int) = bindView()
+                override fun unLiked(likeButton: LikeButton?) {
+                    onUnLike(favorites[bindingAdapterPosition].productID, bindingAdapterPosition)
```
Comment (MohabElziny) [inline app/src/main/java/com/iti/android/team1/ebuy/ui/savedItems/adapter/SavedRecyclerAdapter.kt:40]: favorites[bindingAdapterPosition] is duplicated, Is any suggestions?

**Pull request merged**